core/src/main/scala/neotypes/CypherStringInterpolator.scala

```diff
@@ -17,19 +17,20 @@ object CypherStringInterpolator {
     val params = parameters.iterator
 
     @annotation.tailrec
-    def loop(paramNext: Boolean, output: List[DeferredQueryBuilder.Part]): List[DeferredQueryBuilder.Part] =
+    def loop(paramNext: Boolean, nextSubQuery: Int, acc: List[DeferredQueryBuilder.Part]): List[DeferredQueryBuilder.Part] =
       if (paramNext && params.hasNext) {
         params.next() match {
-          case QueryArg.Param(value)      => loop(paramNext = false, output :+ DeferredQueryBuilder.Param(value))
-          case QueryArg.CaseClass(params) => loop(paramNext = false, output ++ caseClassParts(params))
+          case QueryArg.Param(value)          => loop(paramNext = false, nextSubQuery, acc :+ DeferredQueryBuilder.Param(value))
+          case QueryArg.CaseClass(params)     => loop(paramNext = false, nextSubQuery, acc ++ caseClassParts(params))
+          case QueryArg.QueryBuilder(builder) => loop(paramNext = false, nextSubQuery + 1, acc ++ subQueryParts(builder, index = nextSubQuery))
         }
       } else if (queries.hasNext) {
-        loop(paramNext = true, output :+ DeferredQueryBuilder.Query(queries.next()))
+        loop(paramNext = true, nextSubQuery, acc :+ DeferredQueryBuilder.Query(queries.next(), Seq.empty))
       } else {
-        output
+        acc
       }
 
-    val queryParts = loop(paramNext = false, Nil)
+    val queryParts = loop(paramNext = false, nextSubQuery = 1, Nil)
```

Comment:
Could you please add `acc =` before the `Nil`?
Also, as we discussed the other day, maybe it would be better to use `Vector` due to the constant appending?
While I understand this is run at compile-time and queries are in general _"small"_ so complexity shouldn't matter that much, I wonder if we are introducing unnecessary slowdowns to codebases that have a lot of queries.

Reply:
> Could you please add acc = before the Nil?

Sure

>While I understand this is run at compile-time and queries are in general "small" so complexity shouldn't matter that much, I wonder if we are introducing unnecessary slowdowns to codebases that have a lot of queries.

Perhaps, but IMHO there is no point in optimizing for problems that you don't know exist. I'm also not sure that is a change which would belong in this PR

Reply:
>Perhaps, but IMHO optimizing for problems that you don't know exist is usually an unnecessary time sink. I'm also not sure that is a change which would belong in this PR

I'm a bit tired atm :P I realized it's a really small so I can of course do it if you want. I'm not really sure what's the better option though; using a `mutable.List` or just prepending the parts and reversing the list at the end

Reply:
> using a mutable.List or just prepending the parts instead of appending and then reversing the list at the end

Yeah, there is also another reason why every time I think about this I never actually do anything; there are many ways to tackle this. I just wanted a second opinion about this.
Lets agree that this change doesn't below to this PR and leave the idea in the air in case you want to chat about in in **gitter**.

Reply:
Fine by me :)